### DISCUSSION TITLE
Motr unit benchmark test do panic

### DISCUSSION THREAD

**Discussion by yanqingfu** [Seagate/cortx-motr#607] [Show and tell]
edit the ./scripts/m0 file

![image](https://user-images.githubusercontent.com/68423216/97098290-41e02680-16b6-11eb-8e5f-fb685d498926.png)

> [root@ssc-vm-1061 cortx-motr]# cd scripts
> [root@ssc-vm-1061 scripts]# nano m0
> [root@ssc-vm-1061 scripts]# cd ..
> [root@ssc-vm-1061 cortx-motr]#  ./scripts/m0 run-ut
> ----- run_ut -----
> -- round 1ad-ub[...]adieu-ub[.....]fol-ub[]fom-ub[motr[11960]:  3360  FATAL  [lib/assert.c:50:m0_panic]  panic: (reqh_args->rhia_fid != ((void *)0)) at m0_reqh_init() (reqh/reqh.c:232)  [git: sage-base-1.0-58-gde3ee5d] /var/motr/m0ub/m0trace.11960
> Motr panic: (reqh_args->rhia_fid != ((void *)0)) at m0_reqh_init() reqh/reqh.c:232 (errno: 4) (last failed: none) [git: sage-base-1.0-58-gde3ee5d] pid: 11960  /var/motr/m0ub/m0trace.11960
> /var/cortx/cortx-motr/motr/.libs/libmotr.so.1(m0_arch_backtrace+0x20)[0x7fc104b905a0]
> /var/cortx/cortx-motr/motr/.libs/libmotr.so.1(m0_arch_panic+0xe6)[0x7fc104b90756]
> /var/cortx/cortx-motr/motr/.libs/libmotr.so.1(+0x34f9b4)[0x7fc104b7f9b4]
> /var/cortx/cortx-motr/motr/.libs/libmotr.so.1(+0x3bdd5a)[0x7fc104bedd5a]
> /var/cortx/cortx-motr/ut/.libs/libmotr-ut.so.0(+0x17b22c)[0x7fc0f8fa322c]
> /var/cortx/cortx-motr/motr/.libs/libmotr.so.1(m0_ub_run+0xe1)[0x7fc104b89571]
> /var/cortx/cortx-motr/ut/.libs/lt-m0ub(main+0x2af)[0x4010df]
> /lib64/libc.so.6(__libc_start_main+0xf5)[0x7fc0f844d555]
> /var/cortx/cortx-motr/ut/.libs/lt-m0ub[0x40110f]
> /var/cortx/cortx-motr/utils/m0run: line 416: 11960 Aborted                 (core dumped) $(srcdir_path_of $binary) "$@"

**Comment by yanqingfu** [comment]
error break down: in fom-ub

> [root@ssc-vm-1061 cortx-motr]# ./scripts/m0 run-ut -l
> ----- run_ut -l -----
> Available benchmarks:
> ad-ub
> adieu-ub
> fol-ub
> fom-ub
> list-ub
> memory-ub
> thread-ub
> time-ub
> timer-ub
> tlist-ub
> trace-ub
> varr-ub
> [root@ssc-vm-1061 cortx-motr]# ./scripts/m0 run-ut -t ad-ub
> ----- run_ut -t ad-ub -----
> -- round 1ad-ub[...]
>                bench: [   iter]    min    max    avg   std   sec/op   op/sec block_size time_in_sec    bytes      KiB   BW_KiB     IOPS
> set: ad-ub
>          write-prime: [      1]   0.02   0.02   0.02  0.00% 1.804e-02/5.542e+01     00   0.02         0000     00     00     00
>                write: [    100]   1.41   1.41   1.41  0.00% 1.413e-02/7.076e+01     00   1.41         0000     00     00     00
>                 read: [    100]   0.05   0.05   0.05  0.00% 5.262e-04/1.900e+03     00   0.05         0000     00     00     00
> [root@ssc-vm-1061 cortx-motr]# ./scripts/m0 run-ut -t adieu-ub
> ----- run_ut -t adieu-ub -----
> -- round 1adieu-ub[.....]
>                bench: [   iter]    min    max    avg   std   sec/op   op/sec block_size time_in_sec    bytes      KiB   BW_KiB     IOPS
> set: adieu-ub
>          write-prime: [      1]   0.00   0.00   0.00  0.00% 1.200e-04/8.333e+03   4096   0.00        16384     16 133333   8192
>                write: [    100]   0.00   0.00   0.00  0.00% 2.825e-05/3.540e+04   4096   0.00      1638400   1600 566572  34816
>                 read: [    100]   0.00   0.00   0.00  0.00% 2.720e-05/3.676e+04   4096   0.00      1638400   1600 588235  35840
>           iovec-sort: [ 100000]   0.04   0.04   0.04  0.00% 3.762e-07/2.658e+06   4096   0.04   1638400000 1600000 42533960 2658304
>         iovec-sort-i: [ 100000]  24.30  24.30  24.30  0.00% 2.430e-04/4.115e+03   4096  24.30   1638400000 1600000  65841   4096
> You have new mail in /var/spool/mail/root
> [root@ssc-vm-1061 cortx-motr]# ./scripts/m0 run-ut -t fol-ub
> ----- run_ut -t fol-ub -----
> -- round 1fol-ub[]
>                bench: [   iter]    min    max    avg   std   sec/op   op/sec block_size time_in_sec    bytes      KiB   BW_KiB     IOPS
> set: fol-ub
> [root@ssc-vm-1061 cortx-motr]# ./scripts/m0 run-ut -t fom-ub
> ----- run_ut -t fom-ub -----
> -- round 1fom-ub[motr[26647]:  60a0  FATAL  [lib/assert.c:50:m0_panic]  panic: (reqh_args->rhia_fid != ((void *)0)) at m0_reqh_init() (reqh/reqh.c:232)  [git: sage-base-1.0-58-gde3ee5d] /var/motr/m0ub/m0trace.26647
> Motr panic: (reqh_args->rhia_fid != ((void *)0)) at m0_reqh_init() reqh/reqh.c:232 (errno: 0) (last failed: none) [git: sage-base-1.0-58-gde3ee5d] pid: 26647  /var/motr/m0ub/m0trace.26647
> /var/cortx/cortx-motr/motr/.libs/libmotr.so.1(m0_arch_backtrace+0x20)[0x7f5ab4c885a0]
> /var/cortx/cortx-motr/motr/.libs/libmotr.so.1(m0_arch_panic+0xe6)[0x7f5ab4c88756]
> /var/cortx/cortx-motr/motr/.libs/libmotr.so.1(+0x34f9b4)[0x7f5ab4c779b4]
> /var/cortx/cortx-motr/motr/.libs/libmotr.so.1(+0x3bdd5a)[0x7f5ab4ce5d5a]
> /var/cortx/cortx-motr/ut/.libs/libmotr-ut.so.0(+0x17b22c)[0x7f5aa909b22c]
> /var/cortx/cortx-motr/motr/.libs/libmotr.so.1(m0_ub_run+0xe1)[0x7f5ab4c81571]
> /var/cortx/cortx-motr/ut/.libs/lt-m0ub(main+0x2af)[0x4010df]
> /lib64/libc.so.6(__libc_start_main+0xf5)[0x7f5aa8545555]
> /var/cortx/cortx-motr/ut/.libs/lt-m0ub[0x40110f]
> /var/cortx/cortx-motr/utils/m0run: line 416: 26647 Aborted                 (core dumped) $(srcdir_path_of $binary) "$@"
> [root@ssc-vm-1061 cortx-motr]# ./scripts/m0 run-ut -t list-ub
> ----- run_ut -t list-ub -----
> -- round 1list-ub[..]
>                bench: [   iter]    min    max    avg   std   sec/op   op/sec block_size time_in_sec    bytes      KiB   BW_KiB     IOPS
> set: list-ub
>               insert: [ 100000]  22.57  22.57  22.57  0.00% 2.257e-04/4.430e+03     00  22.57         0000     00     00     00
>               delete: [ 100000]  11.29  11.29  11.29  0.00% 1.129e-04/8.858e+03     00  11.29         0000     00     00     00
> [root@ssc-vm-1061 cortx-motr]# ./scripts/m0 run-ut -t memory-ub
> ----- run_ut -t memory-ub -----
> -- round 1memory-ub[........]
>                bench: [   iter]    min    max    avg   std   sec/op   op/sec block_size time_in_sec    bytes      KiB   BW_KiB     IOPS
> set: memory-ub
>          alloc-small: [5000000]   0.84   0.84   0.84  0.00% 1.671e-07/5.983e+06     00   0.84         0000     00     00     00
>           free-small: [5000000]   0.42   0.42   0.42  0.00% 8.376e-08/1.194e+07     00   0.42         0000     00     00     00
>         alloc-medium: [5000000]   0.69   0.69   0.69  0.00% 1.380e-07/7.247e+06     00   0.69         0000     00     00     00
>          free-medium: [5000000]   0.42   0.42   0.42  0.00% 8.480e-08/1.179e+07     00   0.42         0000     00     00     00
>          alloc-large: [5000000]   2.37   2.37   2.37  0.00% 4.731e-07/2.114e+06     00   2.37         0000     00     00     00
>           free-large: [5000000]   1.03   1.03   1.03  0.00% 2.053e-07/4.871e+06     00   1.03         0000     00     00     00
>           alloc-huge: [   5000]   0.39   0.39   0.39  0.00% 7.857e-05/1.273e+04     00   0.39         0000     00     00     00
>            free-huge: [   5000]   0.15   0.15   0.15  0.00% 2.963e-05/3.375e+04     00   0.15         0000     00     00     00
> [root@ssc-vm-1061 cortx-motr]# ./scripts/m0 run-ut -t thread-ub
> ----- run_ut -t thread-ub -----
> -- round 1thread-ub[...]
>                bench: [   iter]    min    max    avg   std   sec/op   op/sec block_size time_in_sec    bytes      KiB   BW_KiB     IOPS
> set: thread-ub
>                spawn: [  10000]   0.26   0.26   0.26  0.00% 2.622e-05/3.814e+04     00   0.26         0000     00     00     00
>                 join: [  10000]   0.07   0.07   0.07  0.00% 7.382e-06/1.355e+05     00   0.07         0000     00     00     00
>           spawn-init: [  10000]   0.41   0.41   0.41  0.00% 4.084e-05/2.449e+04     00   0.41         0000     00     00     00
> [root@ssc-vm-1061 cortx-motr]# ./scripts/m0 run-ut -t time-ub
> ----- run_ut -t time-ub -----
> -- round 1time-ub[.]
>                bench: [   iter]    min    max    avg   std   sec/op   op/sec block_size time_in_sec    bytes      KiB   BW_KiB     IOPS
> set: time-ub
>                  now: [16777216]   0.56   0.56   0.56  0.00% 3.330e-08/3.003e+07     00   0.56         0000     00     00     00
> [root@ssc-vm-1061 cortx-motr]# ./scripts/m0 run-ut -t timer-ub
> ----- run_ut -t timer-ub -----
> -- round 1timer-ub[....................]
>                bench: [   iter]    min    max    avg   std   sec/op   op/sec block_size time_in_sec    bytes      KiB   BW_KiB     IOPS
> set: timer-ub
>         S-<init>-fin: [      1]   0.00   0.00   0.00  0.00% 5.800e-05/1.724e+04     00   0.00         0000     00     00     00
>         H-<init>-fin: [      1]   0.00   0.00   0.00  0.00% 1.600e-05/6.250e+04     00   0.00         0000     00     00     00
>         S-init-<fini: [      1]   0.00   0.00   0.00  0.00% 3.100e-05/3.226e+04     00   0.00         0000     00     00     00
>         H-init-<fini: [      1]   0.00   0.00   0.00  0.00% 1.000e-06/1.000e+06     00   0.00         0000     00     00     00
>         S-init-<star: [      1]   0.00   0.00   0.00  0.00% 1.000e-06/1.000e+06     00   0.00         0000     00     00     00
>         H-init-<star: [      1]   0.00   0.00   0.00  0.00% 3.000e-06/3.333e+05     00   0.00         0000     00     00     00
>         S-init-start: [      1]   0.00   0.00   0.00  0.00% 1.200e-05/8.333e+04     00   0.00         0000     00     00     00
>         H-init-start: [      1]   0.00   0.00   0.00  0.00% 1.000e-06/1.000e+06     00   0.00         0000     00     00     00
>         S-init-<star: [      1]   0.00   0.00   0.00  0.00% 1.700e-05/5.882e+04     00   0.00         0000     00     00     00
>         H-init-<star: [      1]   0.00   0.00   0.00  0.00% 2.000e-06/5.000e+05     00   0.00         0000     00     00     00
>         S-<init-star: [      1]   0.00   0.00   0.00  0.00% 2.700e-05/3.704e+04     00   0.00         0000     00     00     00
>         H-<init-star: [      1]   0.00   0.00   0.00  0.00% 3.000e-06/3.333e+05     00   0.00         0000     00     00     00
>         S-<init-star: [      1]   0.00   0.00   0.00  0.00% 5.100e-05/1.961e+04     00   0.00         0000     00     00     00
>         H-<init-star: [      1]   0.00   0.00   0.00  0.00% 4.000e-06/2.500e+05     00   0.00         0000     00     00     00
>         S-init-<star: [      1]   0.00   0.00   0.00  0.00% 6.100e-05/1.639e+04     00   0.00         0000     00     00     00
>         H-init-<star: [      1]   0.00   0.00   0.00  0.00% 1.400e-05/7.143e+04     00   0.00         0000     00     00     00
>         S-init-start: [      1]   0.00   0.00   0.00  0.00% 5.100e-05/1.961e+04     00   0.00         0000     00     00     00
>         H-init-start: [      1]   0.00   0.00   0.00  -nan% 0.000e+00/     inf     00   0.00         0000     00     00     00
>         S-<init-star: [      1]   0.00   0.00   0.00  0.00% 9.000e-05/1.111e+04     00   0.00         0000     00     00     00
>         H-<init-star: [      1]   0.00   0.00   0.00  0.00% 1.300e-05/7.692e+04     00   0.00         0000     00     00     00
> [root@ssc-vm-1061 cortx-motr]# ./scripts/m0 run-ut -t tlist-ub
> ----- run_ut -t tlist-ub -----
> -- round 1tlist-ub[..]
>                bench: [   iter]    min    max    avg   std   sec/op   op/sec block_size time_in_sec    bytes      KiB   BW_KiB     IOPS
> set: tlist-ub
>               insert: [ 100000]  82.32  82.32  82.32  0.00% 8.232e-04/1.215e+03     00  82.32         0000     00     00     00
>               delete: [ 100000]  24.98  24.98  24.98  0.00% 2.498e-04/4.004e+03     00  24.98         0000     00     00     00
> [root@ssc-vm-1061 cortx-motr]# ./scripts/m0 run-ut -t trace-ub
> ----- run_ut -t trace-ub -----
> -- round 1trace-ub[...]
>                bench: [   iter]    min    max    avg   std   sec/op   op/sec block_size time_in_sec    bytes      KiB   BW_KiB     IOPS
> set: trace-ub
>                empty: [5000000]   0.21   0.21   0.21  0.00% 4.115e-08/2.430e+07     00   0.21         0000     00     00     00
>                    8: [5000000]   0.21   0.21   0.21  0.00% 4.125e-08/2.424e+07     00   0.21         0000     00     00     00
>                   64: [5000000]   0.21   0.21   0.21  0.00% 4.263e-08/2.346e+07     00   0.21         0000     00     00     00
> [root@ssc-vm-1061 cortx-motr]# ./scripts/m0 run-ut -t varr-ub
> ----- run_ut -t varr-ub -----
> -- round 1varr-ub[.]
>                bench: [   iter]    min    max    avg   std   sec/op   op/sec block_size time_in_sec    bytes      KiB   BW_KiB     IOPS
> set: varr-ub
>                 varr: [      1] 906.19 906.19 906.19  0.00% 9.062e+02/1.104e-03     00 906.19         0000     00     00     00

**Reply by justinzw** [reply]
@yanqingfu sorry for not following up on this last year but was this resolved or is this issue still open?

**Reply by yanqingfu** [reply]
still open

**Reply by mukul-seagate11** [reply]
@huanghua78 @madhavemuri, please address the query?

**Reply by huanghua78** [reply]
https://github.com/Seagate/cortx-motr/pull/592 is created to fix it.

**Comment by huanghua78** [comment]
I will take care of this issue.

**Reply by huanghua78** [reply]
https://github.com/Seagate/cortx-motr/pull/592

PR is created for this issue.

**Reply by huanghua78** [reply]
https://github.com/Seagate/cortx-motr/pull/592  is merged.

This issue has been fixed.
